options.c

```diff
@@ -1002,25 +1000,38 @@ static void toggle_set_term_title(void *data)
 	set_term_title ^= 1;
 }
 
+const char * const shuffle_names[] = {
+	"off", "tracks", "albums", NULL
+};
+
 static void get_shuffle(void *data, char *buf, size_t size)
 {
-	strscpy(buf, bool_names[shuffle], size);
+	strscpy(buf, shuffle_names[shuffle], size);
 }
 
 static void set_shuffle(void *data, const char *buf)
 {
-	int old = shuffle;
-	if (!parse_bool(buf, &shuffle))
+	int tmp;
+
+	if (!parse_enum(buf, 0, 2, shuffle_names, &tmp))
```

Comment:
While this will work correctly and maintain backwards-compatibility if `1`/`0` are used, it won't work with `true`. If the intent is to maintain backwards compatibility, it should probably be set to `tracks` if it's `true` and `parse_enum` fails.